hypervisor/arch/x86/guest/io_emul.c

```diff
@@ -183,7 +171,70 @@ static void dm_emulate_io_complete(struct acrn_vcpu *vcpu)
 		}
 	}
 }
-#endif
+
+/**
+ * @pre (width == 1U) || (width == 2U) || (width == 4U)
+ */
+static uint32_t pio_default_read(__unused struct acrn_vm *vm, __unused uint16_t addr,
+	size_t width)
+{
+	uint32_t val = 0xffffffffU;
+
+	switch (width) {
+	case 1U:
+		val = 0xffU;
+		break;
+	case 2U:
+		val = 0xffffU;
+		break;
+	case 4U:
+		val = 0xffffffffU;
+		break;
+	default:
+		/* This case is unreachable, this is guaranteed by the design. */
+		break;
+	}
+
+	return val;
+}
+
+static void pio_default_write(__unused struct acrn_vm *vm, __unused uint16_t addr,
+	__unused size_t width, __unused uint32_t v)
+{
+	/* ignore write */
+}
+
+/**
+ * @pre (io_req->reqs.mmio.size == 1U) || (io_req->reqs.mmio.size == 2U) ||
+ *      (io_req->reqs.mmio.size == 4U) || (io_req->reqs.mmio.size == 8U)
+ */
+static int32_t mmio_default_access_handler(struct io_request *io_req,
+	__unused void *handler_private_data)
+{
+	struct mmio_request *mmio = &io_req->reqs.mmio;
+
+	if (mmio->direction == REQUEST_READ) {
+		switch (mmio->size) {
+		case 1U:
+			mmio->value = 0xffUL;
+			break;
+		case 2U:
+			mmio->value = 0xffffUL;
+			break;
+		case 4U:
+			mmio->value = 0xffffffffUL;
+			break;
+		case 8U:
+			mmio->value = 0xffffffffffffffffUL;
```

Comment:
can u define a MACRO for those all 0xF...F ?   It is easy to be confused

Reply:
MACROs are added in new PR, please review.